File: projects/android/Sample/app/src/main/java/com/ezored/sample/ui/activity/base/BaseActivity.java
package com.ezored.sample.ui.activity.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import com.ezored.sample.R;
import com.ezored.sample.ui.fragment.base.BaseFragment;

import org.greenrobot.eventbus.EventBus;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getActivityLayout());
        setContentFragment();
        createAll();
    }

    protected void createAll() {
        // ignore
    }

    protected void setContentFragment() {
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        ft.replace(R.id.fragmentContent, getFragmentInstance());
        ft.commit();
    }

    protected BaseFragment getFragmentInstance() {
        return null;
    }

    protected int getActivityLayout() {
        return R.layout.activity_base;
    }

    protected void onHomeButtonSelected() {
        finish();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem menuItem) {
        if (menuItem.getItemId() == android.R.id.home) {
            onHomeButtonSelected();
        }

        return super.onOptionsItemSelected(menuItem);
    }

    protected boolean registerForEventBus() {
        return false;
    }

    @Override
    public void onStart() {
        super.onStart();

        if (registerForEventBus()) {
            EventBus.getDefault().register(this);
        }
    }

    @Override
    public void onStop() {
        super.onStop();

        if (registerForEventBus()) {
            EventBus.getDefault().unregister(this);
        }
    }

}

File: projects/android/Sample/app/src/main/java/com/ezored/sample/ui/activity/base/BaseActivity.kt
package com.ezored.sample.ui.activity.base

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.ezored.sample.R
import com.ezored.sample.ui.fragment.base.BaseFragment
import org.greenrobot.eventbus.EventBus

open class BaseActivity : AppCompatActivity() {

    protected open val fragmentInstance: BaseFragment?
        get() = null

    protected val activityLayout: Int
        get() = R.layout.activity_base

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(activityLayout)
        setContentFragment()
        createAll()
    }

    protected fun createAll() {
        // ignore
    }

    protected fun setContentFragment() {
        val fm = supportFragmentManager
        val ft = fm.beginTransaction()

        fragmentInstance?.let {
            ft.replace(R.id.fragmentContent, it)
        }

        ft.commit()
    }

    protected fun onHomeButtonSelected() {
        finish()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onHomeButtonSelected()
        }

        return super.onOptionsItemSelected(item)
    }

    protected fun registerForEventBus(): Boolean {
        return false
    }

    override fun onStart() {
        super.onStart()

        if (registerForEventBus()) {
            EventBus.getDefault().register(this)
        }
    }

    override fun onStop() {
        super.onStop()

        if (registerForEventBus()) {
            EventBus.getDefault().unregister(this)
        }
    }
}
